test/parallel/test-buffer-toint.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+{
+  const buf = Buffer.from('123');
+  const num = buf.toInt(10);
+
+  assert.strictEqual(num, 123);
+}
+
+{
+  const buf = Buffer.from('010');
+  const num = buf.toInt(10);
```

Comment:
We need to test the default value
```suggestion
  const num = buf.toInt();
```